app/code/core/Mage/Adminhtml/Block/Catalog/Category/Tree.php

```diff
@@ -170,15 +170,13 @@ public function getMoveUrl()
     public function getTree($parenNodeCategory=null)
     {
            $rootArray = $this->_getNodeJson($this->getRoot($parenNodeCategory));
-        $tree = isset($rootArray['children']) ? $rootArray['children'] : array();
-        return $tree;
+        return isset($rootArray['children']) ? $rootArray['children'] : array();
     }
 
     public function getTreeJson($parenNodeCategory=null)
     {
         $rootArray = $this->_getNodeJson($this->getRoot($parenNodeCategory));
-        $json = Mage::helper('core')->jsonEncode(isset($rootArray['children']) ? $rootArray['children'] : array());
-        return $json;
+        return Mage::helper('core')->jsonEncode(isset($rootArray['children']) ? $rootArray['children'] : array());
```

Comment:
`Mage::helper('core')->jsonEncode(isset($rootArray['children'])` is always `true` since `Mage::helper('core')->jsonEncode(true)` and `Mage::helper('core')->jsonEncode(false)`  return string 'true' and string 'false' when I checked. If this is correct, then the check is useless and can be just `return $rootArray['children']`.

Reply:
Or `return $rootArray['children'] ?? [];`

Reply:
Why remove `Mage::helper('core')->jsonEncode()`? Please check closing bracers for `jsonEncode()` ... its

`return Mage::helper('core')->jsonEncode($rootArray['children'] ?? array());`

Reply:
My bad...I missed the braces.